src/steps/users/create-or-update-contact.ts

```diff
@@ -0,0 +1,56 @@
+import { BaseStep, Field, StepInterface } from '../../core/base-step';
+import { FieldDefinition, RunStepResponse, Step, StepDefinition } from '../../proto/cog_pb';
+
+import { baseOperators } from '../../client/constants/operators';
+import * as util from '@run-crank/utilities';
+
+/**
+ * Note: the class name here becomes this step's stepId.
+ @see BaseStep.getId()
+ */
+export class CreateOrUpdateContact extends BaseStep implements StepInterface {
+
+  protected stepName: string = 'Create or update an Iterable Contact';
+
+  protected stepType: StepDefinition.Type = StepDefinition.Type.ACTION;
+
+  protected stepExpression: string = 'create or update an iterable contact';
+
+  protected expectedFields: Field[] = [{
+    field: 'contact',
+    type: FieldDefinition.Type.MAP,
+    description: 'Where keys represent contact profile field names as represented in the Iterable API (including email, which is not technically a profile field).',
+  }];
+
+  async executeStep(step: Step): Promise<RunStepResponse> {
+    let apiRes: any;
+    const stepData: any = step.getData().toJavaScript();
+    const contact: string = stepData.contact;
+
+    if (!contact.hasOwnProperty('email')) {
+      return this.error('An email address must be provided in order to create an Iterable Contact');
+    }
+
+    const data = {
+      email: contact['email'],
+      dataFields: contact,
```

Comment:
- [ ] I think we will want to remove the `email` property from the `contact` object here.  If it is passed along with `dataFields`, it appears to be stored as a custom profile field, which is not intended.

e.g.
![Screen Shot 2020-01-09 at 6 15 07 PM](https://user-images.githubusercontent.com/3496491/72120218-18633380-330c-11ea-9166-1d85a7f511b6.png)

Reply:
Oh!  And if you remove this, suddenly a check like this would fail:

```yml
- step: The email field on Iterable Contact a@b.c should be a@b.c
```

- [ ] Could be fixed by performing the necessary object manipulation in `ContactAwareMixin::getContactByEmail`